sources/packages/web/src/views/student/financial-aid-application/StudentApplication.vue

```diff
@@ -1,103 +1,151 @@
+    const programYearId = ref();
+    const startApplicationForm = ref({} as VForm);
+    const { checkNullOrEmptyRule } = useRules();
 
-    const formLoaded = async (form: any) => {
-      await formioDataLoader.loadProgramYear(form, PROGRAM_YEAR_DROPDOWN_KEY);
-    };
-    const dialogClosed = () => {
+    onMounted(async () => {
+      const programYearOptionTmp =
+        await ProgramYearService.shared.getProgramYearOptions();
+      programYearOptions.value = programYearOptionTmp.map((yearOption) => {
+        return {
+          title: yearOption.description,
+          value: yearOption.id.toString(),
+        };
+      });
+    });
+
+    const closeDialog = () => {
       showDialog.value = false;
     };
-    const customEventCallback = async (form: any, event: FormIOCustomEvent) => {
+    const startApplication = async () => {
       try {
-        const programYearId = formioUtils.getComponentValueByKey(
-          form,
-          PROGRAM_YEAR_DROPDOWN_KEY,
-        );
-        const createDraftResult =
-          await ApplicationService.shared.createApplicationDraft({
-            programYearId: programYearId,
-            data: {},
-            associatedFiles: [],
-          });
-        if (createDraftResult.draftAlreadyExists) {
-          showDialog.value = true;
+        const validationResult = await startApplicationForm.value.validate();
+        if (!validationResult.valid) {
           return;
         }
-        const programYear =
-          await ProgramYearService.shared.getActiveProgramYear(programYearId);
-        if (
-          programYear &&
-          programYear.formName &&
-          FormIOCustomEventTypes.RouteToStartFullTimePartTimeApplication ===
-            event.type
-        ) {
-          router.push({
-            name: StudentRoutesConst.DYNAMIC_FINANCIAL_APP_FORM,
-            params: {
-              selectedForm: programYear.formName,
-              programYearId: programYearId,
-              id: Number(createDraftResult.draftId),
-            },
-          });
+        if (programYearId.value) {
+          const programYearNumberId = Number(programYearId.value);
+          const createDraftResult =
+            await ApplicationService.shared.createApplicationDraft({
+              programYearId: programYearNumberId,
+              data: {},
+              associatedFiles: [],
+            });
+          if (createDraftResult.draftAlreadyExists) {
+            showDialog.value = true;
+            return;
+          }
+          const programYear =
+            await ProgramYearService.shared.getActiveProgramYear(
+              programYearNumberId,
+            );
+          if (programYear) {
+            router.push({
+              name: StudentRoutesConst.DYNAMIC_FINANCIAL_APP_FORM,
+              params: {
+                selectedForm: programYear.formName,
+                programYearId: programYearNumberId,
+                id: Number(createDraftResult.draftId),
```

Comment:
do we need this Parse

Reply:
This one was already there. Will check. Thanks!